impersonate/user.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2021 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package impersonate
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"golang.org/x/oauth2"
+)
+
+func user(ctx context.Context, c Config, client *http.Client) (oauth2.TokenSource, error) {
+	// Default to the longest acceptable value of one hour as the token will
+	// be refreshed automatically if not set.
+	lifetime := 3600 * time.Second
+	if c.Lifetime != 0 {
+		lifetime = c.Lifetime
+	}
```

Comment:
Feels like this should be consolidated before `user` is called. Can add an `autoRefresh` (or similar) argument to control if the token should auto-refresh.

Reply:
Done.